tests/DataFixtures/ORM/PromotionsFixture.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Tests\BBC\ProgrammesPagesService\DataFixtures\ORM;
+
+use BBC\ProgrammesPagesService\Data\ProgrammesDb\Entity\Promotion;
+use DateTime;
+use Doctrine\Common\DataFixtures\AbstractFixture;
+use Doctrine\Common\DataFixtures\DependentFixtureInterface;
+use Doctrine\Common\Persistence\ObjectManager;
+
+class PromotionsFixture extends AbstractFixture implements DependentFixtureInterface
+{
+    private $manager;
+
+    public function getDependencies()
+    {
+        return [
+            MongrelsFixture::Class,
+            ImagesFixture::class,
+        ];
+    }
+
+    public function load(ObjectManager $manager)
+    {
+        $this->manager = $manager;
+
+        $this->buildPromotion(
+            'p000000h',
+            'active promotion of CoreEntity',
+            $this->getReference('b00swyx1'), // promotion of Core entity - serie 1
+            1, // weight
+            true,
+            new DateTime('1900-01-01'),
+            new DateTime('3000-01-01')
+        );
+
+        $this->buildPromotion(
+            'p000001h',
+            'active promotion of Image ',
+            $this->getReference('mg000003'), // promotion of image
+            2,
+            true,
+            new DateTime('1900-01-01'),
+            new DateTime('3000-01-01')
+        );
+
+        $this->buildPromotion(
+            'p000002h',
+            'expired promotion B',
+            $this->getReference('b00swyx1'),
+            3,
+            true,
+            new DateTime('1900-01-01'),
+            new DateTime('2000-01-01')
+        );
+
+        $this->buildPromotion(
+            'p000003h',
+            'disabled promotion C',
+            $this->getReference('b00swyx1'),
+            4,
+            false,
+            new DateTime('1900-01-01'),
+            new DateTime('2000-01-01')
+        );
+
+        $this->buildSuperPromotion(
+            'p000004h',
+            'active super promotion D',
+            $this->getReference('b010t150'), // promotion of Core entity - serie 2
+            5,
+            true,
+            new DateTime('1900-01-01'),
+            new DateTime('3000-01-01')
+        );
+
+        $manager->flush();
+    }
+
+    private function buildPromotion(string $promoPid, string $title, $promotionOf, int $weighting, bool $isActive, DateTime $startDate, DateTime $endDate)
```

Comment:
I think this might be a bit easier to follow if there was a single builder function, and you passed in the value of CascadesToDescendants and the item it is attached too.